lalrpop/src/lr1/codegen/parse_table.rs

```diff
@@ -852,7 +852,11 @@ impl<'ascent, 'grammar, W: Write> CodeGenerator<'ascent, 'grammar, W, TableDrive
                 rust!(
                     self.out,
                     "{} => match {}token {{",
-                    indices.iter().map(|(index, _)| index).format(" | "),
```

Comment:
I guess clippy made a suggestion on the generated code, motivating this change?

It looks like the logic is equivalent iff the series of indices is ordered and always increasing by one.  It should be ordered, since it's the result of `.enumerate()`, but can't it possibly not be increasing by one if the None case occurs on line 823 in between two instances of the Some() case?

Reply:
Yes, clippy is more aggressively linting on or patterns of incrementally increasing values.

That's an interesting point, is there a small test case that helps trigger this? All of the current tests are of the logically equivalent form.

Reply:
Ok, nervermind, found one in lrgrammar.rs

Reply:
I've attempted a fix for this to check the ordering first.

Reply:
The fix looks right.  The weird case is exactly 1 terminal, which looks like it ends up using the inclusive range `N..=N`, instead of the old `N`.  That's probably a really uncommon case though, and an inclusive range of length one is just weird, not wrong.

Is this worth the extra code complexity?  Is it just about making clippy happy, or is there a performance benefit to replacing "1 | 2 | 3 | 4" with "1..=4"?

Reply:
It compiles to the exact same code. Entirely just a clippy thing.

Reply:
Well, a maintainer should have a final say, but I feel as though going from 1 fairly straightforward line to 13 lines with a fair amount of logic to parse and think though isn't worth it to make clippy happy in the generated code.  I would vote for just silencing this one to keep the lalrpop code more readable.

Reply:
Certainly up to the maintainer, I can understand the code is more complex than before to handle this case.

However, it's a little disingenuous to say it went from 1 line to 13 lines. For example, the old case is still there but now takes 5 lines because of formatting... that part didn't suddenly become more complex.

Reply:
I'd just call that "lines of code is a bad metric".  But it's a convenient shorthand.  Personally, I think the fact that you've got two cases instead of 1 is an increase in complexity, and the logic in the window closure, particularly with indexing is certainly more involved than a simple map.

It's not a huge difference, but the "make clippy happy" I would call a near zero benefit in this case, since clippy is really analyzing the readability of the generated code, but our level of care about that is near-zero, and our level of care about the readability of the actual lalrpop code is much higher.

Reply:
I agree to @dburgener. I believe clippy suggests it for readability.
But we maintain code generator, not the generated code here. People will read more generator code than generated code. In my opinion, previous one looks more readable.

Reply:
Fair, I've reverted this

Reply:
@Pat-Lafon will putting clippy ignore to line 808 work? 
https://github.com/lalrpop/lalrpop/pull/824/files#diff-f6fa48d7b0092fbf2bbdf4a7413478e304b5b2127fd314a3848b64e022f7aa71R808

Reply:
Sure, lets try that

Reply:
Marking this as resolved